python/python/lancedb/query.py

```diff
@@ -921,3 +924,339 @@ def refine_factor(self, refine_factor: int) -> LanceHybridQueryBuilder:
         """
         self._vector_query.refine_factor(refine_factor)
         return self
+
+
+class AsyncQueryBase(object):
+    def __init__(self, inner: Union[LanceQuery | LanceVectorQuery]):
+        """
+        Construct an AsyncQueryBase
+
+        This method is not intended to be called directly.  Instead, use the
+        [Table.query][] method to create a query.
+        """
+        self._inner = inner
+
+    def where(self, predicate: str) -> AsyncQuery:
+        """
+        Only return rows matching the given predicate
+
+        The predicate should be supplied as an SQL query string.  For example:
+
+        >>> predicate = "x > 10"
+        >>> predicate = "y > 0 AND y < 100"
+        >>> predicate = "x > 5 OR y = 'test'"
+
+        Filtering performance can often be improved by creating a scalar index
+        on the filter column(s).
+        """
+        self._inner.where(predicate)
+        return self
+
+    def select(self, columns: List[str]) -> AsyncQuery:
+        """
+        Choose which columns to return
+
+        This is a shortcut for [select_with_projection][] when no
+        dynamic columns need to be calculated.  See that method for more
+        details on the performance impact of selecting the right columns.
+        """
+        self._inner.select(columns)
+        return self
+
+    def select_with_projection(self, columns: dict[str, str]) -> AsyncQuery:
```

Comment:
Can we use same `select()` api for both function?
`projection` is a very database-eng term that application / ML developers are unfamiliar with.

Reply:
Ok.  I changed them both to `select`.  I agree that's simpler.